hoss/dimension_utilities.py

```diff
@@ -75,8 +76,149 @@ def prefetch_dimension_variables(opendap_url: str, varinfo: VarInfoFromDmr,
 
     logger.info('Variables being retrieved in prefetch request: '
                 f'{format_variable_set_string(required_dimensions)}')
-    return get_opendap_nc4(opendap_url, required_dimensions, output_dir,
-                           logger, access_token, config)
+
+    required_dimensions_nc4 = get_opendap_nc4(opendap_url,
+                                              required_dimensions, output_dir,
+                                              logger, access_token, config)
+
+    # Create bounds variables if necessary.
+    add_bounds_variables(required_dimensions_nc4, required_dimensions,
+                         varinfo, logger)
+
+    return required_dimensions_nc4
+
+
+def add_bounds_variables(dimensions_nc4: str,
+                         required_dimensions: Set[str],
+                         varinfo: VarInfoFromDmr,
+                         logger: Logger) -> None:
+    """ Augment a NetCDF4 file with artificial bounds variables for each
+        dimension variable that is edge-aligned and does not already
+        have bounds variables.
```

Comment:
Do you mean that I should say we're making these bounds by adding the CF override to `hoss_config.json`, or something more directly in `earthdata-varinfo`?

Reply:
I think the latter - i.e., `Augment a NetCDF4 file with artificial bounds variables for each
        dimension variable that is edge-aligned and does not already
        have bounds variables.`  +  `"edge-aligned" and "has-bounds" are defined by earthdata-varinfo attributes, which come from the netcdf4 variable attributes, or if necessary by overrides in the configuration file`

Reply:
I might slightly disagree with some of the text above. `edge-aligned` doesn't appear in any NetCDF-4 file, because it's something we've coined (and really wish was in the NetCDF-4 file). Also (yet another nit) the metadata attribute for bounds is just `bounds` not `has-bounds`.

Reply:
But the `cell_alignment`/`edge-aligned` attribute only exists in `hoss_config.json`, right? Does that make it an `earthdata-varinfo` attribute? Like, 

"Augment a NetCDF4 file with artificial bounds variables for each dimension variable that has  been identified by `hoss_config.json` to have an `edge-aligned` attribute"?

Or instead of `hoss_config.json`, we say "`earthdata-varinfo` configuration file" as a catch-all for all applications?

(I may over-complicating this, I'm sorry!)